Products/CMFPlone/browser/favicon.py

```diff
@@ -18,9 +20,16 @@ def __init__(self, context, request):
         settings = registry.forInterface(ISiteSchema, prefix="plone")
         if getattr(settings, 'site_favicon', False):
             filename, data = b64decode_file(settings.site_favicon)
-            data = NamedImage(data=data, filename=filename)
+            mimetype = getattr(settings, 'site_favicon_mimetype', "image/vnd.microsoft.icon")
+            data = NamedImage(data=data, contentType=mimetype, filename=filename)
             self.data = data
             self.filename = filename
+        else:
+            basedir = os.path.dirname(os.path.dirname(__file__))
+            with open(os.path.join(basedir, "skins", "plone_images", "default-favicon.ico"), "rb") as icon:
+                data = NamedImage(data=icon.read(), contentType="image/vnd.microsoft.icon", filename="favicon.ico")
+            self.data = data
+            self.filename = "favicon.ico"
```

Comment:
I am not sure I like this fallback.

First it reuses the skins folder (and we are trying to move away from that, see ).
Second, it uses that in a way that cannot be overwritten TTW.

Reply:
These are reasonable objections to which I'd like to answer.

1. We are in fact reusing content from the skins folder, *but we are not using the skins folder or infrastructure here*.  When we finally delete all skins, all we need to do is simply move the single icon to an assets folder somewhere else in the product.  We don't at all depend in this PR on the soon-to-be-removed skins feature.
2. `favicon.ico` can certainly be overwritten through the Web.  All you have to do is upload the icon to the Site Control Panel in Plone.  The infrastructure is already checked in for that — adding that feature was not my call or work, but I was in agreement with those PRs.
3. Additionally, this PR removes the `favicon.ico` on every folder that was (in Plone 5) obtained from the skin via Acquisition, so now any and every folder (other than the INavigationRoot folders) can have `favicon.ico` overwritten through the Web.

Please ack.

Reply:
What I meant is that it cannot be overwritten changing it in the skins folder, I know that this is a fallback ;)

I am fine removing the old `favicon.ico` from the skins folder I have no problem with that.

I am not fine renaming the favicon inside the skins folder and using that with a low level `open` call.

I still have to ponder what could be a better approach, but for the moment I do not like this approach.

Reply:
CC @plone/framework-team do you have an opinion on this one?

Reply:
I would gladly contemplate alternatives.  That said, at some level, the data for the favicon must be opened from disk, then retrieved, then served.  How that is done specifically, I am happy to change if a better alternative is presented.

One thing I know almost for certain: since we are eliminating skins, we should probably not invest any effort in accessing any data using skins-related mechanisms.

Reply:
I am no longer part of the FWT, but I also dislike this fallback solution... If the favicon is not in the settings, then I think it should return None. If a favicon has to be returned, I would rather have a b64 encoded Plone logo instead of a hard-coded path in the filesystem. 
Also, why is the default mt `image/vnd.microsoft.icon` when the field default defined in https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/interfaces/controlpanel.py#L1012 is `image/x-icon` ?

Reply:
@frapell 

> Also, why is the default mt image/vnd.microsoft.icon when the field default defined in https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/interfaces/controlpanel.py#L1012 is image/x-icon ?

Because image/x-icon is not really a thing.  The registered MIME type for Microsoft ICO format is `image/vnd.microsoft.icon`.  I can fix the default in that line if you want.

>  If the favicon is not in the settings, then I think it should return None.

Favicon data has to be returned.

> I would rather have a b64 encoded Plone logo instead of a hard-coded path in the filesystem.

Happy to.  I only read the icon file because it was already checked in.  Where should I check in the base64-encoded data?